storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/operations/get/PreparedGetObject.java

```diff
@@ -0,0 +1,237 @@
+
+import rx.Observable;
+
+import static com.pushtorefresh.storio.internal.Checks.checkNotNull;
+
+/**
+ * Created by ilyazorin on 17/11/15.
+ */
+public class PreparedGetObject<T> extends PreparedGet<T> {
+
+    @NonNull
+    private final Class<T> type;
+
+    @Nullable
+    private final GetResolver<T> explicitGetResolver;
+
+    PreparedGetObject(@NonNull StorIOSQLite storIOSQLite,
+                             @NonNull Class<T> type,
+                             @NonNull Query query,
+                             @Nullable GetResolver<T> explicitGetResolver) {
+        super(storIOSQLite, query);
+        this.type = type;
+        this.explicitGetResolver = explicitGetResolver;
+    }
+
+    PreparedGetObject(@NonNull StorIOSQLite storIOSQLite,
+                             @NonNull Class<T> type,
+                             @NonNull RawQuery rawQuery,
+                             @Nullable GetResolver<T> explicitGetResolver) {
+        super(storIOSQLite, rawQuery);
+        this.type = type;
+        this.explicitGetResolver = explicitGetResolver;
+    }
+
+    /**
+     * Executes Get Operation immediately in current thread.
+     * <p>
+     * Notice: This is blocking I/O operation that should not be executed on the Main Thread,
+     * it can cause ANR (Activity Not Responding dialog), block the UI and drop animations frames.
+     * So please, call this method on some background thread. See {@link WorkerThread}.
+     *
+     * @return single instance of mapped result. Can be null, if no items are found.
+     */
+    @Nullable
+    @SuppressWarnings({"ConstantConditions", "NullableProblems"})
+    @WorkerThread
+    public T executeAsBlocking() {
+        try {
+            final GetResolver<T> getResolver;
+
+            if (explicitGetResolver != null) {
+                getResolver = explicitGetResolver;
+            } else {
+                final SQLiteTypeMapping<T> typeMapping = storIOSQLite.internal().typeMapping(type);
+
+                if (typeMapping == null) {
+                    throw new IllegalStateException("This type does not have type mapping: " +
+                            "type = " + type + "," +
+                            "db was not touched by this operation, please add type mapping for this type");
+                }
+
+                getResolver = typeMapping.getResolver();
+            }
+
+            final Cursor cursor;
+
+            if (query != null) {
+                cursor = getResolver.performGet(storIOSQLite, query);
+            } else if (rawQuery != null) {
+                cursor = getResolver.performGet(storIOSQLite, rawQuery);
+            } else {
+                throw new IllegalStateException("Please specify query");
+            }
+
+            try {
+                final int count = cursor.getCount();
+
+                if (count == 0) {
+                    return null;
+                }
+
+                cursor.moveToNext();
```

Comment:
@geralt-encore nope! You're returning `null` if `cursor.getCount == 0`, but then you've to move cursor to first position and we have small chance of problem in case when `cursor` reports that it have some items but then couldn't move to the position.

Though looks like we're not checking it in other places in such manner, so, feel free to leave this code as is!